src/Nethermind/Nethermind.Consensus/Processing/RecoverSignature.cs

```diff
@@ -102,6 +107,43 @@ public void RecoverData(Block block)
                     if (_logger.IsTrace) _logger.Trace($"Recovered {blockTransaction.SenderAddress} sender for {blockTransaction.Hash}");
                 }
             }
+
+            if (releaseSpec.IsAuthorizationListEnabled)
+            {
+                void RecoverAuthority(AuthorizationTuple tuple)
+                {
+                    Span<byte> msg = stackalloc byte[128];
+                    msg[0] = Eip7702Constants.Magic;
+                    RlpStream rlpStream = _authorizationTupleDecoder.EncodeWithoutSignature(tuple.ChainId, tuple.CodeAddress, tuple.Nonce);
+                    rlpStream.Data.AsSpan().CopyTo(msg.Slice(1));
+                    tuple.Authority = _ecdsa.RecoverAddress(tuple.AuthoritySignature, Keccak.Compute(msg.Slice(0, rlpStream.Data.Length + 1)));
+                }
+
+                foreach (Transaction tx in block.Transactions.AsSpan())
```

Comment:
Should we recover authority in TxPool and try to get them from there? Similar as with Hash and Sender?
Adding @benaadams